test/integration/graphql_test.lua

```diff
@@ -2147,3 +2147,225 @@ function g.test_non_finite_float()
             query_schema, nil, nil, {variables = variables})
     end
 end
+
+g.test_arguments_nullability = function ()
+    local Nullable = true
+    local NonNullable = false
+    local test_enum = types.enum({
+        name = 'My_Enum',
+        values = {
+            a = { value = 'a' },
+            b = { value = 'b' },
+        },
+    })
+
+    local graphql_types = {
+        ['enum'] = { graphql_type = test_enum, var_type = 'My_Enum' },
+        ['boolean'] = { graphql_type = types.boolean, var_type = 'Boolean' },
+        ['id'] = { graphql_type = types.id, var_type = 'ID' },
+        ['int'] = { graphql_type = types.int, var_type = 'Int' },
+        ['float'] = { graphql_type = types.float, var_type = 'Float' },
+        ['string'] = { graphql_type =types.string, var_type = 'String' },
+    }
+
+    local function gen_schema(type_list)
+        local function make_type(graphql_type)
+            if #graphql_type > 1 and not graphql_type[2] then
+                return types.nonNull(graphql_types[graphql_type[1]].graphql_type)
+            else
+                return graphql_types[graphql_type[1]].graphql_type
+            end
+        end
+        local function gen_fields(_type_list)
+            local args = {}
+            for k, v in ipairs(_type_list) do
+                args['arg'..tostring(k)] = make_type(v)
+            end
+            return args
+        end
+
+        return {
+            ['test'] = {
+                kind = types.object({
+                    name = 'result',
+                    fields = gen_fields(type_list)
+                }),
+                arguments = gen_fields(type_list),
+                resolve = function(_, args)
+                    return args
+                end,
+            }
+        }
+    end
+
+    local function gen_query(type_list)
+        local variables = ''
+
+        for k, v in ipairs(type_list) do
+            if v[4] ~= nil then
+                if variables == '' then
+                    variables = variables .. '('
+                end
+
+                local var = '$var'..tostring(k)..': '..graphql_types[v[1]].var_type
+                if v[4] == NonNullable then
+                    var = var .. '!'
+                end
+
+                variables = variables ..var
+            end
```

Comment:
nit: Probably it's better to use table.insert to collect all parts and table.concat(..., '\n') to form graphql query.

Reply:
TBD later if it's have any reason to do in test

Reply:
Skipping this because it seems to be redundant for tests